homeassistant/components/wled/config_flow.py

```diff
@@ -0,0 +1,125 @@
+"""Config flow to configure the AdGuard Home integration."""
+import logging
+from typing import Any, Dict, Optional
+
+import voluptuous as vol
+from wled import WLED, WLEDConnectionError
+
+from homeassistant.components.wled.const import DOMAIN  # pylint: disable=W0611
+from homeassistant.config_entries import (
+    CONN_CLASS_LOCAL_POLL,
+    SOURCE_ZEROCONF,
+    ConfigFlow,
+)
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME
+from homeassistant.helpers import ConfigType
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WLEDFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a WLED config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by the user."""
+        return await self._handle_config_flow(user_input)
+
+    async def async_step_zeroconf(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle zeroconf discovery."""
+        if user_input is None:
+            return self.async_abort(reason="connection_error")
```

Comment:
Is connection_error the correct error when there is missing user_input?

Reply:
In theory, this doesn't happen. Since there is always data according to code logic. However, the typing upstream tell otherwise.

Mypy requires me to handle the None case, which I do now. Without data, I cannot connect, so that seems a fitting error, in this case, to abort with.

This is not user-facing as well, it just prevents the "device" from being discovered as such.

Reply:
I now realize I don't have a unit test for this. I'll add one.